javascript/modules/client/src/internal/utils/plugins.ts

```diff
@@ -1,18 +1,27 @@
 import { ProposalStatus } from "../interfaces/common";
 
-
-export function getProposalStatus(startDate: Date, endDate: Date, executed: boolean, yes?: bigint, no?: bigint): ProposalStatus {
+export function computeProposalStatus(
+  startDate: Date,
+  endDate: Date,
+  executed: boolean,
+  yes?: bigint,
+  no?: bigint,
+): ProposalStatus {
   const now = new Date();
   if (startDate >= now) {
-    return ProposalStatus.PENDING
+    return ProposalStatus.PENDING;
   } else if (endDate >= now) {
-    return ProposalStatus.ACTIVE
+    return ProposalStatus.ACTIVE;
   } else if (executed) {
-    return ProposalStatus.EXECUTED
+    return ProposalStatus.EXECUTED;
   } else if (yes && no && yes > no) {
-    return ProposalStatus.SUCCEEDED
+    return ProposalStatus.SUCCEEDED;
   } else {
-    return ProposalStatus.DEFEATED
+    return ProposalStatus.DEFEATED;
   }
 }
 
+export function isProposalId(propoosalId: string): boolean {
+  const regex = new RegExp(/^0x[A-Fa-f0-9]{40}_0x[A-Fa-f0-9]{1,}$/i);
```

Comment:
Is the `0x1234_0x2345` format what we receive from subgraph?

Reply:
yes, the proposal id is composed this way `daoAddtess_proposalIndex", so for example, the proposal number 2 of the dao `0x1234...` will be `ox1234..._0x2`

Reply:
Ok, then we need a task in Subgraph to make it be just the proposal's ID... otherwise URL's are going to be too long